File: eva-core/eva-core-common/src/main/java/org/pkaq/core/annotation/NoRepeatSubmit.java
package org.pkaq.core.annotation;

import java.lang.annotation.*;

/**
 * @author PKAQ
 */ // 作用到方法上
@Target(ElementType.METHOD)
// 运行时有效
@Retention(RetentionPolicy.RUNTIME)
@Inherited
public @interface NoRepeatSubmit {

}

File: eva-core/src/main/kotlin/io/nerv/core/annotation/NoRepeatSubmit.kt
package io.nerv.core.annotation

import java.lang.annotation.Inherited
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

// 作用到方法上
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER) // 运行时有效
@Retention(RetentionPolicy.RUNTIME)
@Inherited
annotation class NoRepeatSubmit 